components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7516,14 +7476,14 @@ private void handleException(String msg, Throwable t) throws APIManagementExcept
     }
 
     private Optional<APIProductIdentifier> getProductIdIfExists(ResultSet rs) throws SQLException {
-        String productProvider = rs.getString("API_PRODUCT_PROVIDER");
+        String productProvider = rs.getString("API_TYPE");
```

Comment:
This seems to be a bug, assigning API_TYPE to provider variable

Reply:
Ah no. Previously what it did was check whether the API_PRODUCT_PROVIDER column is there to identify whether it is a product or not. Now we use the Type column to check this

Reply:
But you are assigning the value to the variable and using it below to create the APIProductIdentifier

Reply:
ohh . thanks.